src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,160 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> node = new Node<>(tail, value, null);
+        if (tail == null) {
+            head = node;
+        } else {
+            tail.next = node;
+        }
+        tail = node;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexBetween(index);
+        if (index == size) {
+            add(value);
+        } else {
+            link(value, getNode(index));
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        if (list == null) {
+            return;
+        }
+
+        for (T item : list) {
+            add(item);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndexWithin(index);
+        return getNode(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndexWithin(index);
+        Node<T> node = getNode(index);
+        T oldValue = node.value;
+        node.value = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndexWithin(index);
+        Node<T> node = getNode(index);
+        T oldValue = node.value;
+        unlink(node);
+        return oldValue;
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> node = head;
+        while (node != null) {
+            if (node.value == object || (node.value != null && node.value.equals(object))) {
+                unlink(node);
+                return true;
+            }
+            node = node.next;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void checkIndexWithin(int index) {
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException("LOL, YOU DUMB? XDXDXDXDXDX");
+        }
+    }
+
+    private void checkIndexBetween(int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Well, shit happens!");
```

Comment:
include index and size in the message, make it clear what's wrong

Reply:
Oh gosh, i've completely forgot to remove those stupid messages, when i was working on the code implementation 🙈